### PR TITLE
Upgrade to Pants 1.14.0

### PR DESCRIPTION
**Pull request by benjyw**
Update ci config and pants script.

### CODE REVIEW

File: .github/workflows/pants.yaml [hunk 1]
```diff
@@ -18,17 +18,34 @@ jobs:
...
-        path: |
-          ~/.cache/pants/setup
-          ~/.cache/pants/lmdb_store
-          ~/.cache/pants/named_caches
-        key: ${{ runner.os }}-
+        pants-python-version: ${{ matrix.python-version }}
+        # cache0 makes it easy to bust the cache if needed
+        # just increase the integer to start with a fresh cache
+        gha-cache-key: cache0-py${{ matrix.python-version }}
+        # The Kotlin backend doesn't use named caches. See the example-python repo for an example
```
Comment (benjyw) [inline .github/workflows/pants.yaml:36]: Is this statement true?

Comment (tdyas) [reply .github/workflows/pants.yaml:36]: Probably not. While the Kotlin backend does not, the JVM support library that it uses does:

```
$ git grep append_only_caches src/python/pants/backend/kotlin | wc -l
       0
$ git grep append_only_caches src/python/pants/backend/kotlin src/python/pants/jvm | wc -l
       7
```

**Review by tdyas — APPROVED** [review]

File: .github/workflows/pants.yaml (hunk 1 above)
Comment (benjyw) [reply .github/workflows/pants.yaml:36]: Hmm, so what should we be mixing into the cache keys here? In Python it's the lockfiles.

Comment (benjyw) [reply .github/workflows/pants.yaml:36]: I think append_only_caches is distinct from named_caches? I've just realized I'm not sure any more

Comment (tdyas) [reply .github/workflows/pants.yaml:36]: `append_only_caches` is the actual name of what "named caches" are. They are named (append-only) caches.

**Pull request merged**